Fix livetests

jspaaks opened the pull request:
**List of related issues or pull requests**

Refs: -

**Describe the changes made in this pull request**

Livetests were broken because they referenced `Config`. Also in this PR:

1. Renamed `livetests/test_howfairis.py` to `livetests/test_howfairis_on_rsd_urls.py`
1. Removed `fallback_branch` when getting default branch from platform, it's better to fail early than to guess the branch
1. Added an `if-else` to making the request for the default branch, you should only get it if the user didnt specify their own branch. This may help cut down on the number of requests we make. Updated the tests that instantiate `Repo` with `branch` accordingly.
1. Changed the testing setup to a parameterized fixture for each url. This way, you can get output like:
    ```
    platform linux -- Python 3.6.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
    rootdir: ~/github/fair-software/howfairis, configfile: setup.cfg
    plugins: requests-mock-1.8.0, cov-2.11.1
    collected 10 items

    livetests/test_howfairis_on_rsd_urls.py ..........
    ```
    Instead of just 1 pass/fail/error for all urls together.
1. Updated Repo's docstring
1. Testing matrix has been reduced to just Python 3.9 on Linux until we fix :#93:
1. Added section in `README.dev.rst` on how to run `prospector` and `isort`
1. Added a git pre-commit hook for running `prospector` and `isort` exactly as the `README.dev.rst` says
1. Added 2 new Exceptions specific to `howfairis`
1. Changed sphinx theme so the max-wdth of the content is a bit wider. This makes it easier to see full location of e.g. `howfairis.exceptions.howfairis_exception.HowfairisException` without line breaks

**Instructions to review the pull request**

1. activate virtual env
1. `pytest livetests/test_howfairis_on_rsd_urls.py`

livetests status on this branch:

![livetests](https://github.com/fair-software/howfairis/workflows/livetests/badge.svg?branch=fix-livetests)

jspaaks commented on howfairis/repo.py:
```diff
@@ -89,13 +89,12 @@ def _derive_raw_url_format_string(self):
 
     def _get_default_branch(self):
-        fallback_branch = 'main'
         # GitHub API and GitLab API work the same
```
i removed this because failing early is better

fdiblen reviewed (approved):
Reviewed the changes and tested locally. Everything seems to be working. 
Thanks @jspaaks

jspaaks commented:
(Also fixed building the API docs, now merging)

Problems were related to
1. Pygments language `shell` directive not working if you have unbalanced single quotes. Changed to `console` 
2. importing / overriding CSS has recently changed in Sphinx, now you need to use `html_css_files` instead of `html_content` in `docs/conf.py`
